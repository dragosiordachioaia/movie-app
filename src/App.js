import React, { Component } from "react";
import "./App.css";

import axios from "axios";
import { ENDPOINTS, REQUEST_PARAMS } from "./constants";

axios.defaults.params = REQUEST_PARAMS;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      genres: null,
      movies: null,
      config: null,
    };

    this.fetchData = this.fetchData.bind(this);
    this.getImagePath = this.getImagePath.bind(this);
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    axios.get(ENDPOINTS.config).then(({ data }) => {
      this.setState({ config: data.images });
    });
    axios.get(ENDPOINTS.nowPlaying).then(({ data }) => {
      this.setState({ movies: data.results }, () => {});
    });
    axios.get(ENDPOINTS.movieGenres).then(({ data }) => {
      this.setState({ genres: data.genres });
    });
  }

  getImagePath(imageID) {
    if (!this.state.config) {
      return null;
    }

    return `${this.state.config.base_url}${
      this.state.config.poster_sizes[4]
    }${imageID}`;
  }

  render() {
    if (!this.state.config || !this.state.movies || !this.state.genres) {
      return <p>Loading...</p>;
    }
    let posterPath = this.state.movies[0].poster_path;
    let fullPosterPath = this.getImagePath(posterPath);

    return (
      <div className="App">
        <img src={fullPosterPath} />
      </div>
    );
  }
}

export default App;
