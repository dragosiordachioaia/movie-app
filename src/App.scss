@import "./variables";

html {
  box-sizing: border-box;
}
*,
*:before,
*:after {
  box-sizing: inherit;
}

body {
  background-color: $color-primary;
  font-family: "Josefin Sans", sans-serif;
}

.app {
  height: 100vh;
  display: flex;
  flex-direction: column;

  .main-content {
    display: flex;
    height: calc(100vh - #{$header-height});
    flex-direction: row;
  }
}
