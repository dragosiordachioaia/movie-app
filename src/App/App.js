import React, { Component } from "react";
import "./App.scss";

import axios from "axios";
import { ENDPOINTS, REQUEST_PARAMS } from "../constants";

import Header from "Header/Header";
import SideBar from "SideBar/SideBar";
import MovieList from "MovieList/MovieList";

axios.defaults.params = REQUEST_PARAMS;

/**
  The App component is responsible for maintaining the entire state of the app,
  which means that all state-changing operations happen here.
**/

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      genres: null,
      movies: null,
      config: null,
      minRating: 3,
      nameFilter: "",
      sidebarExpanded: false,
    };

    this.fetchData = this.fetchData.bind(this);
    this.onGenreCheck = this.onGenreCheck.bind(this);
    this.onToggleSidebar = this.onToggleSidebar.bind(this);
    this.onUpdateNameFilter = this.onUpdateNameFilter.bind(this);
    this.onMinRatingChange = this.onMinRatingChange.bind(this);
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    axios.get(ENDPOINTS.config).then(({ data }) => {
      this.setState({ config: data.images });
    });
    axios.get(ENDPOINTS.nowPlaying).then(({ data }) => {
      this.setState({ movies: data.results }, () => {});
    });
    axios.get(ENDPOINTS.movieGenres).then(({ data }) => {
      this.setState({ genres: data.genres });
    });
  }

  onGenreCheck(data) {
    let newGenres = JSON.parse(JSON.stringify(this.state.genres));
    newGenres.forEach(genreData => {
      if (genreData.id === data.id) {
        genreData.checked = !genreData.checked;
      }
    });
    this.setState({ genres: newGenres });
  }

  onToggleSidebar() {
    this.setState({ sidebarExpanded: !this.state.sidebarExpanded });
  }

  onUpdateNameFilter(newValue) {
    this.setState({ nameFilter: newValue });
  }

  onMinRatingChange(newMinRating) {
    this.setState({ minRating: newMinRating });
  }

  render() {
    if (!this.state.config || !this.state.movies || !this.state.genres) {
      return <p>Loading...</p>;
    }

    return (
      <div
        className={`app ${
          this.state.sidebarExpanded ? "sidebar-expanded" : ""
        }`}
      >
        <Header onNameFilterChange={this.onUpdateNameFilter} />
        <div className="main-content">
          <SideBar
            genres={this.state.genres}
            onGenreCheck={this.onGenreCheck}
            minRating={this.state.minRating}
            onMinRatingChange={this.onMinRatingChange}
            onExpand={this.onToggleSidebar}
            expanded={this.state.sidebarExpanded}
          />
          <MovieList
            movies={this.state.movies}
            config={this.state.config}
            genres={this.state.genres}
            minRating={this.state.minRating}
            nameFilter={this.state.nameFilter}
          />
        </div>
      </div>
    );
  }
}

export default App;
