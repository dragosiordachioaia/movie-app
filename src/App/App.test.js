import React from "react";
import ReactDOM from "react-dom";
import App from "./App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// import React from 'react';
// import renderer from 'react-test-renderer';
//
// import { CompactInsuranceOrder } from 'new_checkout/components/bundle_parts/compact_insurance_order';
//
// let component;
//
// describe('CompactInsuranceOrder component', () => {
//   it('renders correctly', () => {
//     const props = {
//       order: {
//         event: {
//           content: {
//             full_terms: {
//               value_html: 'terms here',
//             },
//           },
//         },
//       },
//     };
//     const component = renderer.create(<CompactInsuranceOrder {...props} />);
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//   it('calls analytics on T & C click', () => {
//     const props = {
//       order: {
//         event: {
//           content: {
//             full_terms: {
//               value_html: 'terms here',
//             },
//           },
//         },
//       },
//       sendAnalyticsEvent: jest.fn(),
//     };
//     const component = renderer.create(<CompactInsuranceOrder {...props} />);
//     const instance = component.getInstance();
//     instance.onInsuranceTAndCClick();
//     expect(props.sendAnalyticsEvent).toBeCalled();
//   });
// });
