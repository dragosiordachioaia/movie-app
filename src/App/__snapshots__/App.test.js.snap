// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders the expanded state 1`] = `
<div
  className="app sidebar-expanded"
>
  <Header
    onNameFilterChange={[Function]}
  />
  <div
    className="main-content"
  >
    <SideBar
      expanded={true}
      genres={Array []}
      onExpand={[Function]}
      onGenreCheck={[Function]}
      onMinRatingChange={[Function]}
    />
    <MovieList
      config={Object {}}
      genres={Array []}
      movies={Array []}
    />
  </div>
</div>
`;

exports[`correctly renders the non-expanded state 1`] = `
<div
  className="app "
>
  <Header
    onNameFilterChange={[Function]}
  />
  <div
    className="main-content"
  >
    <SideBar
      expanded={false}
      genres={Array []}
      onExpand={[Function]}
      onGenreCheck={[Function]}
      onMinRatingChange={[Function]}
    />
    <MovieList
      config={Object {}}
      genres={Array []}
      movies={Array []}
    />
  </div>
</div>
`;

exports[`renders correctly if loaded 1`] = `
<div
  className="app "
>
  <Header
    onNameFilterChange={[Function]}
  />
  <div
    className="main-content"
  >
    <SideBar
      genres={Array []}
      onExpand={[Function]}
      onGenreCheck={[Function]}
      onMinRatingChange={[Function]}
    />
    <MovieList
      config={Object {}}
      genres={Array []}
      movies={Array []}
    />
  </div>
</div>
`;

exports[`renders loading state initially 1`] = `
<p>
  Loading...
</p>
`;
